File: BeerViewer/Models/kcsapi/kcsapi_mission.cs
﻿namespace BeerViewer.Models.kcsapi
{
	public class kcsapi_mission
	{
		public int api_id { get; set; }
		public string api_disp_no { get; set; }
		public int api_maparea_id { get; set; }
		public string api_name { get; set; }
		public string api_details { get; set; }
		public int api_time { get; set; }
		public int api_difficulty { get; set; }
		public float api_use_fuel { get; set; }
		public float api_use_bull { get; set; }
		public int[] api_win_item1 { get; set; }
		public int[] api_win_item2 { get; set; }
		public int api_return_flag { get; set; }
	}
}

File: WindowFrame/source/System/Interfaces/kcsapi_mission.ts
/** Master information interface for Expedition */
interface kcsapi_mission {
    /** Internal expedition id */
    api_id: number;

    /** Id when display to screen */
    api_disp_no: string;

    /** World tab id */
    api_maparea_id: number;

    /** Expedition name */
    api_name: string;

    /** Expedition description */
    api_details: string;

    /** Seconds to complete */
    api_time: number;

    /** Difficulty, display as `E` ~ `S` */
    api_difficulty: number;

    /** Lost fuels ratio after return (0.0 ~ 1.0) */
    api_use_fuel: number;

    /** Lost ammos ratio after return (0.0 ~ 1.0) */
    api_use_bull: number;

    /** Earnable item `[Id, Count]` */
    api_win_item1: [number, number];

    /** Earnable item `[Id, Count]` */
    api_win_item2: [number, number];

    /** Can return after start? */
    api_return_flag: number;
}
export { // Alias
    kcsapi_mission as kcsapi_mission,
    kcsapi_mission as kcsapi_mst_mission
}